addons/website_event/controllers/main.py

```diff
@@ -312,6 +315,8 @@ def _create_attendees_from_registration_post(self, event, registration_data):
 
     @http.route(['''/event/<model("event.event"):event>/registration/confirm'''], type='http', auth="public", methods=['POST'], website=True)
     def registration_confirm(self, event, **post):
+        if not request.env['ir.http']._verify_request_recaptcha_token('website_event_registration'):
+            raise UserError(_('Suspicious activity detected by Google reCaptcha.'))
```

Comment:
Don't forget to update translation terms so we can get them translated.

- Go in debug mode
- Settings
- Translations
- Export
- New language, PO file, select website_event
- Copy the contents into the existing .pot file
- In the diff only keep new terms, don't update the date or version (no red diff, basically)